exp/ingest/io/processors.go

```diff
@@ -0,0 +1,59 @@
+package io
+
+import (
+	"io"
+
+	"github.com/stellar/go/support/errors"
+)
+
+type ChangeProcessor interface {
+	ProcessChange(change Change) error
+}
+
+type LedgerTransactionProcessor interface {
+	ProcessTransaction(transaction LedgerTransaction) error
+}
+
+func StreamLedgerTransactions(
+	txProcessor LedgerTransactionProcessor,
+	reader LedgerReader,
+) error {
+	for {
+		tx, err := reader.Read()
+		if err == io.EOF {
+			return nil
+		}
+		if err != nil {
+			return errors.Wrapf(err, "could not read transaction")
+		}
+		if err = txProcessor.ProcessTransaction(tx); err != nil {
+			return errors.Wrapf(
+				err,
+				"could not process transaction %v",
+				tx.Index,
+			)
+		}
+	}
+}
+
+func StreamChanges(
```

Comment:
I wonder if `Stream` explains what happens inside properly.
```suggestion
func ProcessChanges(
```